Classes/Private/Manager/Audio/OCTAudioEngine.m

```diff
@@ -0,0 +1,194 @@
+                                           sizeof (enableInput)
+                                           );
+    CheckStatus(status, @"Unable to enable/disable input");
+    return (status == noErr);
+}
+
+-(BOOL)outputEnable:(BOOL)enable error:(NSError **)error
+{
+    return YES;
+}
+
+#pragma mark - Audio Status
+-(BOOL)isAudioRunning
+{
+    Boolean running;
+    CheckStatus(AUGraphIsRunning(processingGraph,
+                                 &running),
+                @"Unable to determine if AUGraph is running");
+    return running;
+}
+
+#pragma mark - Private
+-(BOOL)startAudioSession:(NSError **)error
+{
+    AVAudioSession *session = [AVAudioSession sharedInstance];
+    bool success = [session setCategory:AVAudioSessionCategoryPlayAndRecord
+                   error:error];
+    if (!success) {
+        return NO;
+    }
+    success = [session setActive:YES error:error];
+    if (!success) {
+        return NO;
+    }
+
+    self.sampleRate = session.sampleRate;
+
+    return YES;
+}
+
+-(void)setUpStreamFormat
+{
+    //Always initialize the fields of a new audio stream basic description structure to zero
+    UInt32 bytesPerSample = sizeof (SInt32);
+
+    AudioStreamBasicDescription asbd = {0};
+    asbd.mSampleRate = self.sampleRate;
+    asbd.mFormatID = kAudioFormatLinearPCM;
+    asbd.mFormatFlags = kAudioFormatFlagIsFloat | kAudioFormatFlagsNativeEndian | kAudioFormatFlagIsPacked | kAudioFormatFlagIsNonInterleaved;
+    asbd.mChannelsPerFrame = 2;
+    asbd.mBytesPerFrame = bytesPerSample;
+    asbd.mBitsPerChannel = 8 * bytesPerSample;
+    asbd.mFramesPerPacket = 1;
+    asbd.mBytesPerPacket = bytesPerSample;
+
+    //set the property of the ioUnit's stream format
+    CheckStatus(AudioUnitSetProperty(
+                                           ioUnit,
+                                           kAudioUnitProperty_StreamFormat,
+                                           kAudioUnitScope_Output,
+                                           kInputBus,
+                                           &asbd,
+                                           sizeof(asbd)),
+                @"Error setting Audio Format");
+}
+
+-(BOOL)enableInputOnAudioUnit
+{
+    //set the property of the audio unit to accept input
+    UInt32 enableInput = 1;
+    OSStatus status = AudioUnitSetProperty(
+                                  ioUnit,
+                                  kAudioOutputUnitProperty_EnableIO,//property we are changing
+                                  kAudioUnitScope_Input,
+                                  kInputBus,
+                                  &enableInput,
+                                  sizeof (enableInput)
+                                  );
+    CheckStatus(status, @"Unable to enable input");
+    return (status == noErr);
+}
+
+static void CheckStatus(OSStatus result, NSString *errorString)
+{
+    if(result != noErr)
+    {
+        NSLog(@"%@ Error code: %d", errorString, (int)result);
+    }
+}
```

Comment:
Instead `CheckStatus` method it would be _much_ nicer to have NSError object, that we can pass to end-user. You can use error string that you pass here as an description. See `OCTTox.m` for an example of creating NSError all over place (`createErrorWithCode:description:failureReason:`).